lib/kino/tree.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Kino.Tree do
+  @moduledoc """
+  A kino for interactively viewing nested data as a tree view.
+
+  The data can be any term.
+
+  ## Examples
+
+      data = %{
+        id: 1,
+        email: "user@example.com",
+        inserted_at: ~U[2022-01-01T10:00:00Z],
+        addresses: [
+          %{
+            country: "pl",
+            city: "Kraków",
+            street: "Karmelicka",
+            zip: "00123"
+          }
+        ]
+      }
+
+      Kino.Tree.new(data)
+  """
+
+  use Kino.JS, assets_path: "lib/assets/tree"
+
+  def new(data) do
+    Kino.JS.new(__MODULE__, to_node(data))
+  end
+
+  def to_node(string) when is_binary(string) do
```

Comment:
Let's make this private. For the tests you can define a private function like this:

```elixir
defp tree(%Kino.JS{ref: ref}) do
  send(Kino.JS.DataStore, {:connect, self(), %{origin: "client:#{inspect(self())}", ref: ref}})
  assert_receive {:connect_reply, data, %{ref: ^ref}}
  data
end
```

then we can make the same assertions against the received data :)